pyart/core/grid.py

```diff
@@ -10,39 +10,283 @@
+        self.x = x
+        self.y = y
+        self.z = z
+        self.nx = len(x['data'])
+        self.ny = len(y['data'])
+        self.nz = len(z['data'])
+        if projection is None:
+            self.projection = {
+                'proj': 'pyart_aeqd', '_include_lon_0_lat_0': True}
+        else:
+            self.projection = projection
+
+        self.radar_latitude = radar_latitude
+        self.radar_longitude = radar_longitude
+        self.radar_altitude = radar_altitude
+        self.radar_time = radar_time
+        self.radar_name = radar_name
+        self.nradar = self._find_and_check_nradar()
+
+        # initialize attributes with Lazy load dictionaries
+        self.init_point_x_y_z()
+        self.init_point_longitude_latitude()
+        self.init_point_altitude()
+
+        # Depreciated axes attribute
+        axes = {'time': time,
+                'time_start': time,  # incorrect metadata
+                'time_end': time,    # incorrect metadata
+                'z_disp': z,
+                'y_disp': y,
+                'x_disp': x,
+                'alt': origin_altitude,
+                'lat': origin_latitude,
+                'lon': origin_longitude}
         self.axes = axes
+
         return
 
+    @classmethod
+    def from_legacy_parameters(cls, fields, axes, metadata):
+        """
+        Return a Grid class using legacy parameters.
+
+        Parameters
+        ----------
+        fields : dict
+            Dictionary of field dictionaries.
+        metadata : dict
+            Dictionary of metadata.
+        axes : dict
+            Dictionary of axes dictionaries.
+
+        Returns
+        --------
+        grid : Grid
+            A Grid object.
+
+        """
+        warnings.warn(
+            "from_legacy_parameters is depreciated and will be removed in a " +
+            "future version of Py-ART", DeprecationWarning)
+        time = axes['time']
+        fields = fields
+        metadata = metadata
+        origin_latitude = axes['lat']
+        origin_longitude = axes['lon']
+        origin_altitude = axes['alt']
+        x = axes['x_disp']
+        y = axes['y_disp']
+        z = axes['z_disp']
+        grid = cls(time, fields, metadata,
+                   origin_latitude, origin_longitude, origin_altitude,
+                   x, y, z)
+        return grid
+
+    @property
+    def projection_proj(self):
+        # Proj instance as specified by the projection attribute.
+        # Raises a ValueError if the pyart_aeqd projection is specified.
+        projparams = self.get_projparams()
+        if projparams['proj'] == 'pyart_aeqd':
+            raise ValueError(
+                'Proj instance can not be made for the pyart_aeqd projection')
+        if not _PYPROJ_AVAILABLE:
+            raise MissingOptionalDependency(
+                "Basemap is required to create a Proj instance but it " +
+                "is not installed")
+        proj = pyproj.Proj(projparams)
```

Comment:
Then pyproj.Proj will issue an exception which will not be caught.  Py-ART could catch this exception and offer its own error message, but I think seeing the error from pyproj would be more informative than a "the projection attribute is invalid" message.

I'm not seeing a good way to verify the projection attribute besides trying to use it to create a Proj instance.